src/System.Threading/tests/AsyncLocalTests.cs

```diff
@@ -153,6 +153,60 @@ public static class AsyncLocalTests
             return;
         }
 
+        [Fact]
+        public static void NotifyCountOnObjectValueChange()
+        {
+            var obj0 = new object();
+            var obj1 = new object();
+            var obj2 = new object();
+
+            int asyncLocal0ChangeCount = 0;
+            int asyncLocal1ChangeCount = 0;
+            void VerifyChangeCounts(int expectedAsyncLocal0ChangeCount, int expectedAsyncLocal1ChangeCount)
+            {
+                Assert.Equal(expectedAsyncLocal0ChangeCount, asyncLocal0ChangeCount);
+                Assert.Equal(expectedAsyncLocal1ChangeCount, asyncLocal1ChangeCount);
+            }
+
+            Action<AsyncLocalValueChangedArgs<object>> onAsyncLocal0Changed = e =>
+            {
+                Assert.Null(e.PreviousValue);
+                Assert.Same(obj0, e.CurrentValue);
+                ++asyncLocal0ChangeCount;
+            };
+            VerifyChangeCounts(0, 0);
+            var asyncLocal0 = new AsyncLocal<object>(onAsyncLocal0Changed);
```

Comment:
I don't know exactly how xunit handles invoking the test methods, but it's possible that creating and mutating AsyncLocals here could cause them to bleed across into other tests executed after this.  It'd probably be good to prevent that leaking from happening, by either making the test an async method, by putting the whole body inside of an `ExecutionContext.Run(ExecutionContext.Capture(), _ => { ... }, nul)`, etc.